src/core/compact_object.h

```diff
@@ -536,6 +539,15 @@ class CompactObjectView {
   CompactObj obj_;
 };
 
+class CompactObjTypeConverter {
+ public:
+  static std::string_view ObjTypeToString(CompactObjType type);
+  static CompactObjType ObjTypeFromString(std::string_view sv);
```

Comment:
why use sentinels like kInvalid? optional<unsigned> would've fitted so nicely, given that you use it in ScanOpts

Reply:
I think using sentinels is a valid choice, not worse than optional.

Reply:
Maybe, but using both can be confusing